### ISSUE TITLE
Umbraco Custom Property Validation -> SQLite vs SQL field length issue

### ISSUE THREAD

**Issue by AaronSadlerUK** [umbraco/Umbraco-CMS#16368]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.1.1

### Bug summary

The field length in the database for the regex validation in a document type is not the same between SQL Lite and SQL.
This causes a truncated exception to be thrown when trying to deploy.

### Steps to reproduce

Create a new site (or use a new local Umbraco Cloud) using SQL Lite for the database
Add a textstring property to a document type
Set the validation to custom
Enter the following regex
`^(https?:\\/\\/)?((www|m)\\.)?(youtube\\.com\\/(watch\\?v=|embed\\/|v\\/)|youtu\\.be\\/|youtube\\.[a-z]{2,3}\\/(watch\\?v=|embed\\/|v\\/)|youtube-nocookie\\.com\\/(embed\\/)|vimeo\\.com\\/(channels\\/[A-Za-z0-9]+\\/videos\\/|groups\\/[A-Za-z0-9]+\\/videos\\/|video\\/)|vimeo\\.[a-z]{2,3}\\/(channels\\/[A-Za-z0-9]+\\/videos\\/|groups\\/[A-Za-z0-9]+\\/videos\\/|video\\/))([A-Za-z0-9_-]+)(\\?[A-Za-z0-9=&_-]*)?$`
Deploy to Umbraco Cloud (or to a site using SQL)

### Expected result / actual result

I would expect to be able to use the same length validation regex with SQL as with SQLite

Instead a truncated exception is thrown, the field type should be changed to NTEXT to resolve this

---
_This item has been added to our backlog AB#62164_

**Comment by bergmania** [comment]
Correct, Sqlite do not respect field lengths. IMO this should be fixed by having a max length on the input field

**Comment by bjarnef** [comment]
Should this one be closed as it was tagged for `release/13.6.0`?

**Comment by AndyButland** [comment]
I don't think we have a fix for 15 yet - that's the reason we left this one open.

**Comment by AndyButland** [comment]
Fixed via https://github.com/umbraco/Umbraco-CMS/pull/21175 for 17.1.
